devices/flash-zynq7000/flashcfg.c

```diff
@@ -46,128 +47,270 @@
 #define FLASH_CMD_4SE        0xdc /* 4-byte 64KB sector erase*/
 #define FLASH_CMD_BE         0x60 /* Chip erase */
 
+#define FLASH_TIMEOUT_CMD_MS 1000
+#define FLASH_TIMEOUT_WIP_MS 1000
+
+
+/* dummyCyc == -1, means that it has to be set for each memory separately */
+static const flash_cmd_t flash_defCmds[flash_cmd_end] = {
+	{ FLASH_CMD_RDID, 1, 24, 1 },
+	{ FLASH_CMD_RDSR1, 1, 8, 1 },
+	{ FLASH_CMD_WRDI, 1, 0, 1 },
+	{ FLASH_CMD_WREN, 1, 0, 1 },
+	{ FLASH_CMD_READ, 4, 0, 1 },
+	{ FLASH_CMD_4READ, 5, 0, 1 },
+	{ FLASH_CMD_FAST_READ, 4, -1, 1 },
+	{ FLASH_CMD_4FAST_READ, 5, -1, 1 },
+	{ FLASH_CMD_DOR, 4, -1, 2 },
+	{ FLASH_CMD_4DOR, 5, -1, 2 },
+	{ FLASH_CMD_QOR, 4, -1, 4 },
+	{ FLASH_CMD_4QOR, 5, -1, 4 },
+	{ FLASH_CMD_DIOR, 4, -1, 2 },
+	{ FLASH_CMD_4DIOR, 5, -1, 2 },
+	{ FLASH_CMD_QIOR, 4, -1, 4 },
+	{ FLASH_CMD_4QIOR, 5, -1, 4 },
+	{ FLASH_CMD_PP, 4, 0, 1 },
+	{ FLASH_CMD_4PP, 5, 0, 1 },
+	{ FLASH_CMD_QPP, 4, 0, 4 },
+	{ FLASH_CMD_4QPP, 5, 0, 4 },
+	{ FLASH_CMD_P4E, 4, 0, 1 },
+	{ FLASH_CMD_4P4E, 5, 0, 1 },
+	{ FLASH_CMD_SE, 4, 0, 1 },
+	{ FLASH_CMD_4SE, 5, 0, 1 },
+	{ FLASH_CMD_BE, 1, 0, 1 }
+};
+
+
+static int flashcfg_wren(const flash_info_t *info)
+{
+	int res;
+	flash_cmd_t cmd;
+	u8 tx[3];
+
+	/* Write enable */
+	cmd = info->cmds[flash_cmd_wren];
+	if (cmd.size > sizeof(tx)) {
+		return -EINVAL;
+	}
+
+	hal_memset(tx, 0, cmd.size);
+	tx[0] = cmd.opCode;
+
+	qspi_start();
+	res = qspi_polledTransfer(tx, NULL, cmd.size, FLASH_TIMEOUT_CMD_MS);
+	qspi_stop();
+
+	return res;
+}
 
-static void flashcfg_defaultCmds(flash_info_t *info)
+
+static int flashcfg_statusRegGet(const flash_info_t *info, u8 *val)
 {
-	info->cmds[flash_cmd_rdid].opCode = FLASH_CMD_RDID;
-	info->cmds[flash_cmd_rdid].size = 1;
-	info->cmds[flash_cmd_rdid].dummyCyc = 24;
+	int res;
+	size_t cmdSz;
+	const flash_cmd_t cmd = info->cmds[flash_cmd_rdsr1];
+	u8 rx[2] = { 0, 0 };
+	u8 tx[2] = { 0, 0 };
+
+	cmdSz = cmd.size + (cmd.dummyCyc * cmd.dataLines) / 8;
```

Comment:
Length of the transaction is not due to dummy cycles. It's command -> response, so when there're 0 dummy cycles, it's length will be 2